src/main/java/org/apache/sling/distribution/agent/impl/PrivilegeDistributionRequestAuthorizationStrategy.java

```diff
@@ -68,8 +76,8 @@ public void checkPermission(@NotNull ResourceResolver resourceResolver, @NotNull
     private void checkPermissionForAdd(Session session, String[] paths)
             throws RepositoryException, DistributionException {
         AccessControlManager acMgr = session.getAccessControlManager();
-
-        Privilege[] privileges = new Privilege[]{acMgr.privilegeFromName(jcrPrivilege), acMgr.privilegeFromName(Privilege.JCR_READ)};
+        additionalJcrPrivilegesForAdd = additionalJcrPrivilegesForAdd != null ? additionalJcrPrivilegesForAdd : new String[] {Privilege.JCR_READ};
```

Comment:
If the config does not contain values of additionalJcrPrivilegesForAdd or additionalJcrPrivilegesForDelete (as is the case with default config right now) then we switch to using the earlier defaults, viz., JCR_READ and JCR_REMOVE_NODE, respectively as captured in the code. Do you mean something else?

Reply:
Instead of the switch in the code, I suggest to leverage the default value of the property.

```
@Property(value="jcr:read", ...)
private static final String JCR_ADD_PRIVILEGES = "additionalJcrPrivilegesForAdd";
```

Reply:
@tmaret Thanks. I have made the changes.